students/arun_nalla/session04/Lesson4_Mailroom_Part2.py

```diff
@@ -0,0 +1,74 @@
+#!usr/bin/env python
+
+donors_list = [('Aron King', [1, 2]), ('Becky Patty', [3000, 1000]),
+               ('Charlie Lee', [666, 222]), ('Thomas Dave', [202, 90, 450]), ('Nancy Crow', [100, 200, 300])]
+
+donors_dict = {x: y for x, y in donors_list}
+
+
+def thanks_email(donors_dict):
+    while True:
+        name_input = input("Enter full name of a donor>>>").title()
+        if name_input == 'List':
+            print(list(donors_dict.keys()))
+        else:
+            for name in donors_dict.keys():
+               if name_input == name:
+                   d = {}
+                   d['name'] = name
+                   donation_amount = int(input('Enter donation amount: '))
+                   donors_dict[name].append(donation_amount)
+                   d['donation'] = donation_amount
+                   print("Dear {name},\n\tThank you for generous donation of ${donation:.2f} to ZYX charity\nSincerely,\nZYX Team".format(**d))
+                   return
```

Comment:
Because you are now using a dict, there's no need to use a `for` loop to search for the entered name. You can simply say `if name_input in donors_dict:`, to check if the `donors_dict` contains an entry whose key matches `name_input` value. No `for` loop needed. In other words, you can change your lines 15 and 16 to `if name_input in donors_dict:`, and replace `name` in line 18, 20 & 22 with `name_input`.